packages/examples/kademlia-discovery/src/lib.rs

```diff
@@ -156,7 +156,11 @@ pub async fn observer_node(validator_addr: &str, api_server_port: u16) -> Result
         .add_websockets_server(validator_addr)
         .build(kolme.clone())?;
 
-    set.spawn(gossip.run());
+    set.spawn(async {
+        gossip.run().await;
+        #[allow(unreachable_code)]
+        Err(anyhow::anyhow!("Unexpected exit from gossip"))
+    });
```

Comment:
One disadvantage of changing the return type of the component run functions is that we need to repeat this pattern everywhere they're spawned in a JoinSet that also includes non-diverging tasks, which is kind of ugly.

I could add a helper function like this to kolme::common (or elsewhere):

```rust
pub async fn diverging_task_to_result<F, T>(task: F) -> Result<()>
where
    F: std::future::Future<Output = T> + Send + 'static,
    T: Send + 'static,
{
    task.await;
    panic!("Unpexpected exit from diverging task");
}
```

And then this would change to:

```rust
set.spawn(diverging_task_to_result(gossip.run()));
```

Worth doing?